### PR TITLE
Add inheritsPermissionsFrom to unifiedroledefinition

### PR DESCRIPTION
**Pull request by abhijeetsinha**
https://microsoftgraph.visualstudio.com/onboarding/_git/onboarding?path=%2Freviews%2F898-expose-inherited-role-definitions%2Fapi.md&version=GBmaster&_a=preview

### CODE REVIEW

File: api-reference/beta/api/unifiedroledefinition-update.md [hunk 1]
```diff
@@ -53,6 +53,7 @@ In the request body, supply the values for relevant fields that should be update
 |resourceScopes|String collection| List of scopes permissions granted by the role definition apply to. Currently only "/" is supported. Read-only when isBuiltIn is true. **DO NOT USE. This property will be deprecated soon. Attach scope to role assignment.**|
 |rolePermissions|[unifiedRolePermission](../resources/unifiedrolepermission.md) collection| List of permissions included in the role. Read-only when isBuiltIn is true. Required. |
 |templateId|String| Custom template identifier that can be set when isBuiltIn is false. This identifier is typically used if one needs an identifier to be the same across different directories. Read-only when isBuiltIn is true. |
+|inheritsPermissionsFrom| [unifiedRoleDefinition](../resources/unifiedroledefinition.md) collection| Read-only collection of role definitions that the given role definition inherits from. So far, only Azure AD built-in roles supports this attribute. |
```
Comment (davidmu1) [inline api-reference/beta/api/unifiedroledefinition-update.md:56]:
```suggestion
|inheritsPermissionsFrom| [unifiedRoleDefinition](../resources/unifiedroledefinition.md) collection| Read-only collection of role definitions that the given role definition inherits from. Only Azure AD built-in roles support this attribute. |
```

File: api-reference/beta/resources/unifiedroledefinition.md [hunk 2]
```diff
@@ -37,6 +37,7 @@ A unifiedRoleDefinition is a collection of permissions listing the operations th
 |resourceScopes|String collection| List of scopes permissions granted by the role definition apply to. Currently only "/" is supported. Read-only when isBuiltIn is true. **DO NOT USE. This is going to be deprecated soon. Attach scope to role assignment** | 
 |rolePermissions|[unifiedRolePermission](unifiedrolepermission.md) collection| List of permissions included in the role. Read-only when isBuiltIn is true. Required. |
 |templateId|String| Custom template identifier that can be set when isBuiltIn is false. This identifier is typically used if one needs an identifier to be the same across different directories. Read-only when isBuiltIn is true. |
+|inheritsPermissionsFrom| [unifiedRoleDefinition](unifiedroledefinition.md) collection| Read-only collection of role definitions that the given role definition inherits from. So far, only Azure AD built-in roles supports this attribute. |
```
Comment (davidmu1) [inline api-reference/beta/resources/unifiedroledefinition.md:40]:
```suggestion
|inheritsPermissionsFrom| [unifiedRoleDefinition](unifiedroledefinition.md) collection| Read-only collection of role definitions that the given role definition inherits from. Only Azure AD built-in roles support this attribute. |
```

File: api-reference/beta/api/unifiedroledefinition-get.md [hunk 3]
```diff
@@ -115,6 +117,260 @@ Content-type: application/json
...
+                "microsoft.azure.serviceHealth/allEntities/allTasks",
+                "microsoft.azure.supportTickets/allEntities/allTasks",
+                "microsoft.office365.serviceHealth/allEntities/allTasks",
+                "microsoft.office365.supportTickets/allEntities/allTasks",
+                "microsoft.office365.webPortal/allEntities/standard/read"
+            ],
+            "condition": null
+        }
+    ],
+    "inheritsPermissionsFrom@odata.context": "https://graph.microsoft.com/beta/$metadata#roleManagement/directory/roleDefinitions('fdd7a751-b60b-444a-984c-02652fe8fa1c')/inheritsPermissionsFrom",
```
Comment (davidmu1) [inline api-reference/beta/api/unifiedroledefinition-get.md:305]: "inheritsPermissionsFrom@odata.context" seems odd to me. Should it be "inheritsPermissionsFrom"?

File: api-reference/beta/api/unifiedroledefinition-get.md [hunk 4]
```diff
@@ -115,6 +117,260 @@ Content-type: application/json
...
```
Comment (davidmu1) [inline api-reference/beta/api/unifiedroledefinition-get.md:211]: "inheritsPermissionsFrom@odata.context" seems odd to me. Should it be "inheritsPermissionsFrom"?

File: api-reference/beta/api/rbacapplication-post-roledefinitions.md [hunk 5]
```diff
@@ -134,7 +134,9 @@ Content-type: application/json
             ],
             "condition": null
         }
-    ]
+    ],
+    "inheritsPermissionsFrom@odata.context": "https://graph.microsoft.com/beta/$metadata#roleManagement/directory/roleDefinitions('c2cb59a3-2d01-4176-a458-95b0e674966f')/inheritsPermissionsFrom",
```
Comment (davidmu1) [inline api-reference/beta/api/rbacapplication-post-roledefinitions.md:138]: "inheritsPermissionsFrom@odata.context" seems odd to me. Should it be "inheritsPermissionsFrom"?

File: api-reference/beta/api/rbacapplication-list-roledefinitions.md [hunk 6]
```diff
@@ -123,6 +123,12 @@ Content-type: application/json
                     ],
                     "condition": null
                 }
+            ],
+            "inheritsPermissionsFrom@odata.context": "https://graph.microsoft.com/beta/$metadata#roleManagement/directory/roleDefinitions('729827e3-9c14-49f7-bb1b-9608f156bbb8')/inheritsPermissionsFrom",
```
Comment (davidmu1) [inline api-reference/beta/api/rbacapplication-list-roledefinitions.md:127]: "inheritsPermissionsFrom@odata.context" seems odd to me. Should it be "inheritsPermissionsFrom"?

File: api-reference/beta/api/rbacapplication-list-roledefinitions.md [hunk 7]
```diff
@@ -144,6 +150,12 @@ Content-type: application/json
                     ],
                     "condition": null
                 }
+            ],
+            "inheritsPermissionsFrom@odata.context": "https://graph.microsoft.com/beta/$metadata#roleManagement/directory/roleDefinitions('f023fd81-a637-4b56-95fd-791ac0226033')/inheritsPermissionsFrom",
```
Comment (davidmu1) [inline api-reference/beta/api/rbacapplication-list-roledefinitions.md:154]: "inheritsPermissionsFrom@odata.context" seems odd to me. Should it be "inheritsPermissionsFrom"?

File: api-reference/beta/api/rbacapplication-list-roledefinitions.md [hunk 8]
```diff
@@ -167,6 +179,12 @@ Content-type: application/json
                     ],
                     "condition": null
                 }
+            ],
+            "inheritsPermissionsFrom@odata.context": "https://graph.microsoft.com/beta/$metadata#roleManagement/directory/roleDefinitions('b0f54661-2d74-4c50-afa3-1ec803f12efe')/inheritsPermissionsFrom",
```
Comment (davidmu1) [inline api-reference/beta/api/rbacapplication-list-roledefinitions.md:183]: "inheritsPermissionsFrom@odata.context" seems odd to me. Should it be "inheritsPermissionsFrom"?

File: api-reference/beta/api/unifiedroledefinition-get.md [hunk 9]
```diff
@@ -51,9 +51,11 @@ Do not supply a request body for this method.
 
 If successful, this method returns a `200 OK` response code and the requested [unifiedRoleDefinition](../resources/unifiedroledefinition.md) object in the response body.
 
-## Example
+## Examples
 
-### Request
+### Example 1: Get definition of a custom role
```
Comment (davidmu1) [inline api-reference/beta/api/unifiedroledefinition-get.md:56]:
```suggestion
### Example 1: Get the definition of a custom role
```

File: api-reference/beta/api/unifiedroledefinition-get.md [hunk 10]
```diff
@@ -115,6 +117,260 @@ Content-type: application/json
             ],
             "condition": null
         }
+    ],
+    "inheritsPermissionsFrom@odata.context": "https://graph.microsoft.com/beta/$metadata#roleManagement/directory/roleDefinitions('f189965f-f560-4c59-9101-933d4c87a91a')/inheritsPermissionsFrom",
+    "inheritsPermissionsFrom": []
+}
+```
+
+### Example 2: Get definition of a built-in role
```
Comment (davidmu1) [inline api-reference/beta/api/unifiedroledefinition-get.md:126]:
```suggestion
### Example 2: Get the definition of a built-in role
```

File: api-reference/beta/api/unifiedroledefinition-get.md [hunk 11]
```diff
@@ -115,6 +117,260 @@ Content-type: application/json
...
+    ],
+    "inheritsPermissionsFrom@odata.context": "https://graph.microsoft.com/beta/$metadata#roleManagement/directory/roleDefinitions('fdd7a751-b60b-444a-984c-02652fe8fa1c')/inheritsPermissionsFrom",
+    "inheritsPermissionsFrom": [
+        {
+            "id": "88d8e3e3-8f55-4a1e-953a-9b9898b8876b"
+        }
+    ]
+}
+```
+### Example 3: Get definition of an Azure AD built-in role and $expand on the role it inherits from
```
Comment (davidmu1) [inline api-reference/beta/api/unifiedroledefinition-get.md:219]:
```suggestion
### Example 3: Get the definition of a built-in role and expand on the role it inherits from
```

File: api-reference/beta/api/rbacapplication-post-roledefinitions.md (hunk 5 above)
Comment (abhijeetsinha) [reply api-reference/beta/api/rbacapplication-post-roledefinitions.md:138]: @davidmu1 - Its correct. Here's the snippet of actual response - 

image

File: api-reference/beta/resources/unifiedroledefinition.md (hunk 2 above)
Comment (abhijeetsinha) [reply api-reference/beta/resources/unifiedroledefinition.md]: Fixed

File: api-reference/beta/api/unifiedroledefinition-update.md (hunk 1 above)
Comment (abhijeetsinha) [reply api-reference/beta/api/unifiedroledefinition-update.md]: Fixed.

File: api-reference/beta/api/unifiedroledefinition-get.md (hunk 3 above)
Comment (abhijeetsinha) [reply api-reference/beta/api/unifiedroledefinition-get.md:305]: It's correct. It's a navigation property.

File: api-reference/beta/api/unifiedroledefinition-get.md (hunk 11 above)
Comment (abhijeetsinha) [reply api-reference/beta/api/unifiedroledefinition-get.md]: Fixed

File: api-reference/beta/api/unifiedroledefinition-get.md (hunk 4 above)
Comment (abhijeetsinha) [reply api-reference/beta/api/unifiedroledefinition-get.md:211]: Its correct. it's a navigation property.

File: api-reference/beta/api/unifiedroledefinition-get.md (hunk 10 above)
Comment (abhijeetsinha) [reply api-reference/beta/api/unifiedroledefinition-get.md]: Fixed

File: api-reference/beta/api/unifiedroledefinition-get.md (hunk 9 above)
Comment (abhijeetsinha) [reply api-reference/beta/api/unifiedroledefinition-get.md]: Fixed.

File: api-reference/beta/api/rbacapplication-list-roledefinitions.md (hunk 6 above)
Comment (abhijeetsinha) [reply api-reference/beta/api/rbacapplication-list-roledefinitions.md:127]: Its correct. It's a navigation property.

**Comment by abhijeetsinha** [comment]
@Lauragra - I have set it to 'ready to merge' yesterday, but it hasn't been merged yet. Is there any update needed in my PR?

**Comment by Lauragra** [comment]
> @Lauragra - I have set it to 'ready to merge' yesterday, but it hasn't been merged yet. Is there any update needed in my PR?

I was OOF for a couple of days. I will take a look and get it merged in as soon as I can.

File: api-reference/beta/api/unifiedroledefinition-get.md [hunk 12]
```diff
@@ -115,6 +117,260 @@ Content-type: application/json
...
+
+#### Request
+
+The following is an example of the request.
+
+
+# [HTTP](#tab/http)
+<!-- {
+  "blockType": "request",
+  "name": "get_unifiedroledefinition"
```
Comment (Lauragra) [inline api-reference/beta/api/unifiedroledefinition-get.md:136]: Each example should have a unique name so it's easy to identify in the validation reports.

File: api-reference/beta/api/unifiedroledefinition-get.md [hunk 13]
```diff
@@ -115,6 +117,260 @@ Content-type: application/json
...
+}-->
+
+```msgraph-interactive
+GET https://graph.microsoft.com/beta/roleManagement/directory/roleDefinitions/fdd7a751-b60b-444a-984c-02652fe8fa1c
+```
+# [C#](#tab/csharp)
```
Comment (Lauragra) [inline api-reference/beta/api/unifiedroledefinition-get.md:142]: Are these the same SDK snippets are are used in Example 1? Generally, the snippet automation takes care of generating and adding the SDK snippets for new examples. I think you'll want to remove these and after we merge, the snippets will be created and added via the weekly automation.

Comment (abhijeetsinha) [reply api-reference/beta/api/unifiedroledefinition-get.md]: @Lauragra - Removed these. Thanks!

File: api-reference/beta/api/unifiedroledefinition-get.md (hunk 12 above)
Comment (abhijeetsinha) [reply api-reference/beta/api/unifiedroledefinition-get.md]: @Lauragra  - Fixed.

**Pull request merged**